bench.sh

```diff
@@ -457,13 +448,25 @@ do
     --long) LONG=1; shift ;;
     --graphs) GRAPH=1; shift ;;
     --no-measure) MEASURE=0; shift ;;
+    --with-chart-exe) CHART_EXE=$1; shift ;;
     --) shift; break ;;
     -*|--*) print_help ;;
     *) break ;;
   esac
 done
 GAUGE_ARGS=$*
 
+# Parse cabal build flags for any quirks
+# Dont specify the "build" directory of another build.
+# Legal: dist-newstyle, dist-newstyle/commit-hash
+# Illegal: dist-newstyle/build
+for option in "$CABAL_BUILD_FLAGS"
+do
+  case $option in
+    --builddir=*) CABAL_BUILD_DIR=${option:11}; break ;;
```

Comment:
You may want to quote this (`CABAL_BUILD_DIR="${option:11}"`) in case there are spaces in the dir name.